File: src/server/Hoard.Core/Application/Transactions/TransactionWriteDto.cs
namespace Hoard.Core.Application.Transactions;

public class TransactionWriteDto
{
    public int? AccountId { get; set; }
    public int? InstrumentId { get; set; }
    public int? TransactionTypeId { get; set; }
    public DateOnly? Date { get; set; }
    public string? Notes { get; set; }
    public decimal? Units { get; set; }
    
    public string? ContractNoteReference { get; set; }
    
    public decimal? DealingCharge { get; set; }
    public decimal? FxCharge { get; set; }
    public decimal? StampDuty { get; set; }
    public decimal? PtmLevy { get; set; }
    
    public decimal? Price { get; set; }
    public decimal? Value { get; set; }
}

File: src/client/hoard-ui/src/api/dtos/Transactions/TransactionWriteDto.ts
export interface TransactionWriteDto {
  accountId: number | null
  instrumentId: number | null
  transactionTypeId: number | null
  date: string | null // ISO: YYYY-MM-DD
  notes: string | null
  units: number | null
  contractNoteReference: string | null
  dealingCharge: number | null
  fxCharge: number | null
  stampDuty: number | null
  ptmLevy: number | null
  price: number | null
  value: number | null
}
